### ISSUE TITLE
Server Error: 401 Unauthorized.

### ISSUE THREAD

**Issue by Raj-Adarsh-Root** [dgmid/nextcloud-bookmark-manager#19]
Hi @dgmid and @LnxGnome

Quick background details - I am using a **macOS 10.15.5** laptop, and recently downloaded NBM-v2.0.4.

I don't have a personal server, and am using the services of a couple of free NextCloud service providers: 

1.) [emma.cloud.tabdigital.eu](https://emma.cloud.tabdigital.eu/login)
2.) [nextcloud03.webo.hosting](https://nextcloud03.webo.hosting/login)

On trying to connect with these servers, I encountered the following error in both the cases. SS attached:

<p align="center">
<img width="424" alt="Emma" src="https://user-images.githubusercontent.com/67596675/86208381-12009680-bb8e-11ea-82f4-03ea750af147.png">
</p>

<p align="center">
<img width="428" alt="Webo" src="https://user-images.githubusercontent.com/67596675/86208385-13ca5a00-bb8e-11ea-970a-c68c0b826720.png">
</p>

I downgraded the version of NBM to check for compatibility, and used NBM-v2.0.3 and v2.0.2. Same error!

Obviously, since I couldn't login, the _bookmarks.json_ file is showing ```data = "null"```. Any help appreciated.

EDIT: I don't know if this is helpful, but the RSS Feed is: 

```https://emma.cloud.tabdigital.eu/apps/bookmarks/public/rest/v2/bookmark?folder=-1&format=rss&page=-1```

**Comment by marcelklehr** [comment]
It would be helpful to know the version of the server-side bookmarks app.

**Comment by Raj-Adarsh-Root** [comment]
Thank you so much for your reply. I had opened a ticket with [webo.hosting](https://nextcloud03.webo.hosting/), and this is their reply:

```
Hello Raj,

Thank you for your message.

To use RSS URL feed mentioned in your ticket you will have to authenticate with your username and password, that's why you receive Error 401: Unauthorized.

We do not support third-party apps and since I am not familiar with the software you are mentioning, It would be best to reach out to developers of the software. But in any case, if the app does not support authentication you won't be able to connect via RSS feed.

Bookmarks version is 3.2.4
```

**Comment by marcelklehr** [comment]
The auth code looks correct: https://github.com/dgmid/nextcloud-bookmark-manager/blob/master/app-source/js/fetch.js#L90

Do you have strange (non-ASCII) characters in your password perhaps?

**Comment by Raj-Adarsh-Root** [comment]
My username is of the form: ```xyz (at) pm.me ```

And my password is (ASCII) alphanumeric.

**Comment by marcelklehr** [comment]
Are you able to connect to nextcloud bookmarks using other clients, i.e. the android or iOS apps, or floccus?

**Comment by Raj-Adarsh-Root** [comment]
Yes, I am able to connect!

I have connected to my NextCloud Bookmarks using:

1.) Floccus (installed on Firefox, Ubuntu-20.04)
2.) [nextBookmark](https://gitlab.com/altepizza/nextbookmark) (on iOS-13.5.1)

The author @dgmid had mentioned in a previous comment his versioning: https://github.com/dgmid/nextcloud-bookmark-manager/issues/17#issuecomment-627743874

**Comment by marcelklehr** [comment]
mmh, cc @altepizza: any idea?

**Comment by altepizza** [comment]
@marcelklehr hmpf not yet. I believe to remember that basic auth via the mail adress might lead to a 401 but I'm not 100% sure. Gonna test it later.

**Comment by dgmid** [comment]
@Raj-Adarsh-Root @marcelklehr I ran a few tests and can confirm that the 401 occurs with a username that contains an @ symbol. Not sure why though.

**Comment by marcelklehr** [comment]
@dgmid That may have to do with using btoa/atob, in floccus I'm using a separate Base64 library, AFAIK.

**Comment by Raj-Adarsh-Root** [comment]
> @Raj-Adarsh-Root @marcelklehr I ran a few tests and can confirm that the 401 occurs with a username that contains an @ symbol. Not sure why though.

Thank you so much for your reply.

I was just curious, and sorry to bother you with this, but would you be having time to release a patch to fix the problem? Or is it something maybe you plan for the next release? In-case you don't have time, I'll look into ways to change my "Username" :-)

**Comment by dgmid** [comment]
@Raj-Adarsh-Root I've had another look at this and identified the problem - it's a bug in my code. I will try release a patch to this as soon as I am able.

**Comment by Raj-Adarsh-Root** [comment]
Am happy to read that :-) Thank you.

**Comment by dgmid** [comment]
Finally got around to releasing the fix for this bug: [v2.0.5](https://github.com/dgmid/nextcloud-bookmark-manager/releases/tag/v2.0.5)
Sorry for the delay!
